share/availability/light/availability.go

```diff
@@ -76,16 +73,13 @@ func (la *ShareAvailability) SharesAvailable(ctx context.Context, dah *share.Roo
 	defer cancel()
 
 	log.Debugw("starting sampling session", "root", dah.Hash())
-	ses := blockservice.NewSession(ctx, la.bserv)
 	errs := make(chan error, len(samples))
 	for _, s := range samples {
 		go func(s Sample) {
-			root, leaf := ipld.Translate(dah, s.Row, s.Col)
-
-			log.Debugw("fetching share", "root", dah.Hash(), "leaf CID", leaf)
-			_, err := share.GetShare(ctx, ses, root, leaf, len(dah.RowsRoots))
```

Comment:
Ok, I removed the blockservice session because I wasn't able to understand how it was being used and incorrectly assumed it wasn't being used at all. Very interesting, would love to talk about this in sync

Reply:
This will be released in the `v0.18.0` release of boxo.
It has been implemented in ipfs/boxo@22fa8b16a3e530b86542286bebba77c08e17487e